### PR TITLE
feat(ci): pre release feature deploy v2

### PR DESCRIPTION
**Pull request by Snojo**
# ...
fixes: https://github.com/island-is/island.is/issues/8033
linked: https://github.com/island-is/helm/pull/661

Current issue: https://github.com/island-is/island.is/runs/7580316436?check_suite_focus=true#step:8:55
Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -101,8 +118,23 @@ jobs:
...
+            do
+                echo "Checking branch against pre_release_pattern '$pre_pattern'"
+                echo "Check if this is a pre-release and if it should generate a feature-deploy"
+                if [[ "$GIT_BRANCH" =~ $pre_pattern ]]; then
+                  echo "'$GIT_BRANCH' matches 'pre_$pattern', continuing with build"  
+                  echo "::set-output name=PRE_CHECK::feature-deploy"
+                  echo "::set-output name=PRE_RELEASE::true"
+                  exit 0
+                fi
+            done
```
Comment (pshomov) [inline .github/workflows/push.yml:131]: Can we get rid of this duplication?

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -101,8 +118,23 @@ jobs:
...
           echo "::set-output name=PRE_CHECK::push"
 
+      - name: Debug - What is my PRE_CHECK
+        run: |
+          echo "My Pre-check is: '$PRE_CHECK'"
+
```
Comment (pshomov) [inline .github/workflows/push.yml:137]: Remove this?

File: .github/workflows/push.yml [hunk 3]
```diff
@@ -572,6 +620,10 @@ jobs:
           aws-secret-access-key: ${{ secrets.ECR_AWS_SECRET_ACCESS_KEY }}
           aws-region: eu-west-1
 
+      - name: Debug - Affected Services
+        run: |
+          printf "Affected services are: ${{needs.prepare.outputs.IMAGES}}\n"
+
```
Comment (pshomov) [inline .github/workflows/push.yml:626]: Clean this up?

**Review by pshomov — COMMENTED** [review]
LGTM, needs a bit of cleanup

File: .github/workflows/push.yml (hunk 1 above)
Comment (Snojo) [reply .github/workflows/push.yml:131]: Not sure. It was your idea :D 
One of them is checking when we create a branch and the other is checking on any other event, while skipping PR

**Review by Snojo — COMMENTED** [review]
clean up

**Review by pshomov — APPROVED** [review]
LGTM

**Pull request merged**